pkg/templates/k8sresource_funcs.go

```diff
@@ -96,6 +176,49 @@ func (t *TemplateResolver) fromConfigMap(namespace string, cmapname string, key
 	return keyVal, nil
 }
 
+// copies data values in the given Configmap, namespace.
+func (t *TemplateResolver) copyConfigMapData(namespace string, cmapname string) (string, error) {
+	klog.V(2).Infof("copyConfigMapData for namespace: %v, configmap name: %v", namespace, cmapname)
+
+	ns, err := t.getNamespace("copyConfigMapData", namespace)
+	if err != nil {
+		return "", err
+	}
+
+	configmapsClient := (*t.kubeClient).CoreV1().ConfigMaps(ns)
+	configmap, getErr := configmapsClient.Get(context.TODO(), cmapname, metav1.GetOptions{})
+
+	if getErr != nil {
+		if k8serrors.IsNotFound(getErr) {
+			// Add to the referenced objects if the object isn't found since the consumer may want to watch the object
+			// and resolve the templates again once it is present.
+			t.addToReferencedObjects("/v1", "ConfigMap", ns, cmapname)
+		}
+
+		klog.Errorf("Error getting configmap:  %v", getErr)
+		err := fmt.Errorf("failed getting the ConfigMap %s from %s: %w", cmapname, ns, getErr)
+
+		return "", err
+	}
+
+	klog.V(2).Infof("Configmap is %v", configmap)
+
+	data := make(map[string]string)
+
+	for key, val := range configmap.Data {
+		data[key] = val
+	}
+
+	yamlBytes, err := yaml.Marshal(data)
```

Comment:
After thinking about this, I think converting it to JSON is the right move since JSON is a subset of what YAML supports and the result after templates are executed gets converted back to JSON to the caller of `ResolveTemplate` anyway, so it ends up not being seen by the user.